server/schedule/operator_controller.go

```diff
@@ -313,7 +312,17 @@ func (oc *OperatorController) AddWaitingOperator(ops ...*operator.Operator) int
 
 	oc.Unlock()
 	operatorWaitCounter.WithLabelValues(ops[0].Desc(), "promote-add").Inc()
-	oc.PromoteWaitingOperator()
+
+	if added > 0 {
```

Comment:
return as soon as possible so the maintainer needn't keep too many conditions in mind. For example, here we could return when we find added=0 so we needn't keep in mind the condition `add=0` anymore.